### PR TITLE
Fyle filter pipe try1 - Trying to implement filter pipe

### PR DESCRIPTION
**Pull request by shishirams222**

### CODE REVIEW

File: src/app/shared/pipes/filter.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'filter'
+})
+
+export class FilterPipe implements PipeTransform {
+  transform(items: any[], searchText: string): any[] {
```
Comment (abhishek1234321) [inline src/app/shared/pipes/filter.pipe.ts:8]: would it be `any[]` always or can we be more specific? like `string[]` etc?

File: src/app/shared/pipes/filter.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    // if there are no items, return empty list
+		if (!items) {
```
Comment (abhishek1234321) [inline src/app/shared/pipes/filter.pipe.ts:10]: indentation 2 spaces please

Comment (abhishek1234321) [inline src/app/shared/pipes/filter.pipe.ts:10]: what if `items` is an `[]`?

File: src/app/shared/pipes/filter.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+			return [];
+		}
+
+		// if there is no search text, return full list
+		if (!searchText) {
```
Comment (abhishek1234321) [inline src/app/shared/pipes/filter.pipe.ts:15]: do we have any convention wherein some parameters can be optional in a filter?

File: src/app/shared/pipes/filter.pipe.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+			return items; 
+		}
+
+		searchText = searchText.toLowerCase();
```
Comment (abhishek1234321) [inline src/app/shared/pipes/filter.pipe.ts:19]: searchTextLowerCase

File: src/app/shared/pipes/filter.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+		return items.filter ( item => {
+      return item.toLowerCase().includes(searchText);
```
Comment (abhishek1234321) [inline src/app/shared/pipes/filter.pipe.ts:22]: indentation is off

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

File: src/app/shared/pipes/filter.pipe.ts (hunk 1 above)
Comment (shishirams222) [reply src/app/shared/pipes/filter.pipe.ts]: It should be `any[]`, list might not be just list of strings. So kept it as `any[]`. There are more checks inside

File: src/app/shared/pipes/filter.pipe.ts (hunk 2 above)
Comment (shishirams222) [reply src/app/shared/pipes/filter.pipe.ts]: Updated

Comment (shishirams222) [reply src/app/shared/pipes/filter.pipe.ts]: updated

File: src/app/shared/pipes/filter.pipe.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'filter'
+})
+
```
Comment (Julias0) [inline src/app/shared/pipes/filter.pipe.ts:6]: no empty line please

**Pull request closed without merging**